### PR TITLE
Create UI home

### PR DESCRIPTION
**Pull request by ndt0409**
## Related Tickets
- [#Task 55572: Create UI home screen](https://edu-redmine.sun-asterisk.vn/issues/55572)
## WHAT
- Create ImageButton to show characters, stories, series
- Create RecyclerView to show events, creators
- Create item RecyclerView
- Create ImageButton See more (events, creators)
- Remote datasource
## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/res/layout/item_banner.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ImageView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/imageBanner"
```
Comment (hungld-1302) [inline app/src/main/res/layout/item_banner.xml]: image_banner

File: gradle.properties [hunk 2]
```diff
@@ -22,3 +22,6 @@ kotlin.code.style=official
 # thereby reducing the size of the R class for that library
 android.nonTransitiveRClass=true
 android.enableJetifier=true
+API_KEY=INSERT_YOUR_API_KEY_HERE
+HASH=INSERT_YOUR_HASH_HERE
+TS=INSERT_YOUR_TS_HERE
```
Comment (hungvv-1301) [inline gradle.properties:27]: End line

**Review by hungvv-1301 — APPROVED** [review]

**Review by nhanviet1 — APPROVED** [review]

File: app/src/main/java/com/ndt/marvelapp/data/model/ComicResult.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package com.ndt.marvelapp.data.model
+
+import com.google.gson.annotations.SerializedName
+
+class ComicResult(
+    @SerializedName("results")
+    val results: List<Comic>
```
Comment (vietnh-1109) [inline app/src/main/java/com/ndt/marvelapp/data/model/ComicResult.kt]: nhớ check null nha , gson nó parse ko có trường này là crash đó

File: app/src/main/java/com/ndt/marvelapp/data/model/Image.kt [hunk 4]
```diff
@@ -0,0 +1,8 @@
+package com.ndt.marvelapp.data.model
+
+import com.google.gson.annotations.SerializedName
+
+data class Image(
+    @SerializedName("path")
+    val path: String
```
Comment (vietnh-1109) [inline app/src/main/java/com/ndt/marvelapp/data/model/Image.kt]: tương tự

File: app/src/main/java/com/ndt/marvelapp/di/NetworkModule.kt [hunk 5]
```diff
@@ -0,0 +1,41 @@
+package com.ndt.marvelapp.di
+
```
Comment (vietnh-1109) [inline app/src/main/java/com/ndt/marvelapp/di/NetworkModule.kt:2]: remove blank

File: app/src/main/java/com/ndt/marvelapp/di/NetworkModule.kt [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+val networkModule = module {
+    single { provideOkHttpClient() }
+
+    single { provideRetrofit(get()) }
+}
+
+private fun provideOkHttpClient(): OkHttpClient {
+    val builder = OkHttpClient.Builder()
+    builder.addInterceptor { chain ->
+        val original = chain.request()
```
Comment (vietnh-1109) [inline app/src/main/java/com/ndt/marvelapp/di/NetworkModule.kt]: thêm httplogging

File: app/src/main/res/layout/fragment_home.xml [hunk 7]
```diff
@@ -1,9 +1,190 @@
...
+            android:id="@+id/image_button_event_next"
+            android:layout_width="@dimen/dp_50"
+            android:layout_height="@dimen/dp_50"
+            android:layout_marginTop="@dimen/dp_12"
+            android:background="@color/transparent"
+            android:scaleType="center"
+            android:src="@drawable/ic_next"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/text_character" />
+
```
Comment (vietnh-1109) [inline app/src/main/res/layout/fragment_home.xml:132]: remove blank

File: app/src/main/res/layout/item_creator.xml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+        app:layout_constraintDimensionRatio="3:1.75"
+        app:layout_constraintHeight_percent="0.8"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <ImageView
+            android:id="@+id/image_creator"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent" />
+    </androidx.cardview.widget.CardView>
```
Comment (vietnh-1109) [inline app/src/main/res/layout/item_creator.xml]: dùng shapableImage view đi Tú ơi , ko bo ảnh bằng cardview nhé , lại còn để trong item list , nếu nhiều data nó bindview lag lắm

**Pull request merged**